jobs/gorouter/templates/pre-start.erb

```diff
@@ -6,10 +6,33 @@ source /var/vcap/packages/routing_utils/syslog_utils.sh
 
 tee_output_to_sys_log "${LOG_DIR}" "pre-start" <%= p("router.logging.format.timestamp") %>
 
+<%
+    ports = []
+    ports.append(p("router.port")) # has default. will always exist.
+    ports.append(p("router.status.port")) # has default. will always exist.
+    ports.append(p("router.tls_port")) # has default. will always exist.
+
+    if_p('router.prometheus.port') do |port|
+        ports.append(port)
+    end
+
+    # ipv4 only. This makes me sad. I couldn't find a ruby package that takes an IP
+    # with a port and returns just the port.
+    debug_address = p("router.debug_address").split(":").last.to_i
+    if debug_address > 0
+        ports.append(debug_address)
+    end
+
+    ports.sort!
+%>
+
 <% if p('router.set_kernel_parameters') %>
     # Ephemeral port range
     echo "<%= p("router.ip_local_port_range") %>" > /proc/sys/net/ipv4/ip_local_port_range
 
+    # Reserved Ports, not ephemeral
+    echo <%= ports.join(",") %> > /proc/sys/net/ipv4/ip_local_reserved_ports
```

Comment:
Is this really how that works? All the things i'm reading make it seem like a two integer range, not a comma-separated value.

Reply:
> ip_local_reserved_ports - list of comma separated ranges
	Specify the ports which are reserved for known third-party
	applications. These ports will not be used by automatic port
	assignments (e.g. when calling connect() or bind() with port
	number 0). Explicit port allocation behavior is unchanged.
>	The format used for both input and output is a comma separated
	list of ranges (e.g. "1,2-4,10-10" for ports 1, 2, 3, 4 and
	10). Writing to the file will clear all previously reserved
	ports and update the current list with the one given in the
	input.
	
[Per the kernel docs](https://www.kernel.org/doc/Documentation/networking/ip-sysctl.txt).

Reply:
oh i see - different proc setting than the range. 👍 neat!